pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3513,12 +3513,13 @@ int BPF_KPROBE(trace_check_helper_call)
 
     int func_id;
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 13, 0))
```

Comment:
> Sometimes the only way to detect the presence of the necessary feature is through checking the Linux kernel version. Libbpf allows to do that from BPF program code using a special extern variable:

`extern int LINUX_KERNEL_VERSION __kconfig;`

https://nakryiko.com/posts/bpf-core-reference-guide/#linux-kernel-version

Reply:
Actually it seems that the commit that introduced the parameters change of this function also introduced the helper addition (https://github.com/torvalds/linux/commit/cc0f83530934dda0ce1dd01990d2f37f8c3f0d92#diff-edbb57adf10d1ce1fbb830a34fa92712fd01db1fbd9b6f2504001eb7bcc7b9d0R5666) so I think it is good enough. Checking for kernel version doesn't seem to have an advantage here since code can be backported.

Reply:
verifier <= 5.4 had issues when using conditional branch and scalar values coming from kconfig relocations IIRC. we should revisit that, but I remember we couldn't do that that is why we started using type checks for version checks.

Reply:
![image](https://github.com/aquasecurity/tracee/assets/7395852/de6d50f3-c533-41be-9378-3540a9b7dc92)

ok libbpf treats that as a special kconfig file now and treats different distro versioning. IF we can get passed the 5.4 verifier then we would be good to start using that schema for linux kernel versions checking (BUT remember that things are backported to RHEL/SUSE etc, so it would be better to check for types being used).

Reply:
What about a schema using both `LINUX_KERNEL_VERSION` and `bpf_core_*_exists()`? Maybe setting bits for enabled features in a bitmask (or map) populated by that schema.